integration/integration_test.go

```diff
@@ -275,32 +275,100 @@ func TestPublishMessage(t *testing.T) {
 	}
 
 	unique := "UniqueStringToLookForInOutput"
-	sampleValues, err := helpers.NewStruct(map[string]interface{}{
-		"string": unique,
-		"number": 42,
+	events, err := createEvents([]string{unique})
+	if err != nil {
+		t.Fatalf("error creating events: %s\n", err)
+	}
+
+	_, err = client.PublishEvents(env.ctx, &messages.PublishRequest{
+		Events: events,
 	})
+	if err != nil {
+		t.Fatalf("Error publishing event: %s", err)
+	}
+
+	found, err = env.WaitUntil("stdout", unique)
+	if err != nil {
+		t.Fatalf("Error looking for publish results: %s", err)
+	}
+
+	if !found {
+		stderr, err := env.GetStderr()
+		if err != nil {
+			t.Errorf("Error getting standard error: %s", err)
+		}
+		stdout, err := env.GetStdout()
+		if err != nil {
+			t.Errorf("Error getting standard out: %s", err)
+		}
+		t.Fatalf("Event wasn't published.\nstderr:\n%s\nstdout:\n%s\n", stderr, stdout)
+	}
+}
+
+func TestDiskQueue(t *testing.T) {
+	var config strings.Builder
+	queue_path := t.TempDir()
+
+	_, _ = config.WriteString("server:\n")
+	_, _ = config.WriteString("  strict_mode: false\n")
+	_, _ = config.WriteString("  port: 50052\n")
+	_, _ = config.WriteString("  tls: false\n")
+	_, _ = config.WriteString("logging:\n")
+	_, _ = config.WriteString("  level: debug\n")
+	_, _ = config.WriteString("  selectors: [\"*\"]\n")
+	_, _ = config.WriteString("  to_stderr: true\n")
+	_, _ = config.WriteString("queue:\n")
+	_, _ = config.WriteString("  disk:\n")
+	_, _ = config.WriteString("    path: " + queue_path + "\n")
+	_, _ = config.WriteString("    max_size: 10G\n")
+
+	env, err := NewTestingEnvironment(t, config.String())
+	if err != nil {
+		t.Fatalf("Error creating environment: %s", err)
+	}
+	t.Cleanup(func() { env.Stop() })
+
+	found, err := env.WaitUntil("stderr", "gRPC server is ready and is listening on")
+	if err != nil {
+		t.Fatalf("Error waiting for server to start: %s", err)
+	}
+
+	if !found {
+		stderr, err := env.GetStderr()
+		if err != nil {
+			t.Errorf("Error getting standard error: %s", err)
+		}
+		stdout, err := env.GetStdout()
+		if err != nil {
+			t.Errorf("Error getting standard out: %s", err)
+		}
+		t.Fatalf("Test executable failed to start.\nstderr:\n%s\nstdout:\n%s\n", stderr, stdout)
+	}
 
-	e := &messages.Event{
-		Timestamp: timestamppb.Now(),
-		Source: &messages.Source{
-			InputId:  "input",
-			StreamId: "stream",
-		},
-		DataStream: &messages.DataStream{
-			Type:      "log",
-			Dataset:   "default",
-			Namespace: "default",
-		},
-		Metadata: sampleValues,
-		Fields:   sampleValues,
-	}
-	events := []*messages.Event{e}
+	client, err := env.NewClient("localhost:50052")
```

Comment:
A little worried that the hard-coded ports are going to collide with something, but it looks like the rest of the tests already do that, so unless you're particularly interested in modifying the framework to assign random free ports, that's probably a task for another day.

Reply:
yeah it is a risk.  Going to wait and see if it is a problem.  If it is, my guess is we should move to Docker for starting the Shipper.